packages/app-staking/src/Actions/Accounts.tsx

```diff
@@ -2,36 +2,73 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
-import { I18nProps } from '@polkadot/ui-app/types';
+import { ApiProps } from '@polkadot/ui-api/types';
 import { ComponentProps } from '../types';
+import { I18nProps } from '@polkadot/ui-app/types';
+import { KeyringSectionOption } from '@polkadot/ui-keyring/options/types';
+import { SubjectInfo } from '@polkadot/ui-keyring/observable/types';
+import { withApi, withMulti } from '@polkadot/ui-api/with';
 
 import React from 'react';
+import styled from 'styled-components';
 import { Button, CardGrid, Icon } from '@polkadot/ui-app';
 import createOption from '@polkadot/ui-keyring/options/item';
 import { getAddressName } from '@polkadot/ui-app/util';
-import keyring from '@polkadot/ui-keyring';
-import styled from 'styled-components';
 
 import Account from './Account';
-import { KeyringSectionOption } from '@polkadot/ui-keyring/options/types';
 import StartStaking from './NewStake';
 import translate from '../translate';
 
-type Props = I18nProps & ComponentProps;
+type Props = I18nProps & ComponentProps & ApiProps;
 
 type State = {
-  isNewStakeOpen: boolean
+  allAccounts: SubjectInfo | undefined;
+  isNewStakeOpen: boolean,
+  myStashes: Array<string | null> | undefined
 };
 
 class Accounts extends React.PureComponent<Props,State> {
   state: State = {
-    isNewStakeOpen: false
+    allAccounts: undefined,
+    isNewStakeOpen: false,
+    myStashes: []
   };
 
+  componentWillReceiveProps ({ allAccounts, api }: Props) {
+    const previousState = this.state;
+
+    if (allAccounts && allAccounts !== previousState.allAccounts) {
+      api.query.staking.bonded.multi(Object.keys(allAccounts))
+      .then((myControlers) => {
+        const result: string[] = [];
+
+        myControlers.forEach((value,index) => {
+
+          if (value.toString() !== '') {
+            result.push(Object.keys(allAccounts)[index]);
+          }
+        });
+
+        return result;
+      })
+      .then((myStashes) => {
+        this.setState({
```

Comment:
This could be problematic and make React throw a hissy fit - and sadly we have a coup[le of those that slipped in lately. So basically, imagine -

- component is loaded
- query is made
- component is unloaded
- result comes back, then fires and it tries to set the state... on a non-existent component 
- boom!